tests/unit/test-firefox/test.firefox.js

```diff
@@ -263,6 +295,160 @@ describe('firefox', () => {
+             profilesDirPath,
+             'fake-manually-default-profile'
+           );
+
+           await createFakeProfilesIni(profilesDirPath, [
+             {
+               Name: 'default',
+               Path: 'fake-default-profile',
+               IsRelative: 1,
+             },
+             {
+               Name: 'dev-edition-default',
+               Path: 'fake-devedition-default-profile',
+               IsRelative: 1,
+             },
+             {
+               Name: 'manually-set-default',
+               Path: absProfilePath,
+               Default: 1,
+             },
+           ]);
+
+           const isFirefoxDefaultPath = await firefox.isDefaultProfile(
+             path.join(profilesDirPath, 'fake-default-profile'),
+             FakeProfileFinder
+           );
+           assert.equal(
+             isFirefoxDefaultPath, true,
+             'Firefox default profile'
+           );
+
+           const isDevEditionDefaultPath = await firefox.isDefaultProfile(
+             path.join(profilesDirPath, 'fake-devedition-default-profile'),
+             FakeProfileFinder
+           );
+           assert.equal(
+             isDevEditionDefaultPath, true,
+             'Firefox DevEdition default profile'
+           );
+
+           const isManuallyDefault = await firefox.isDefaultProfile(
+             absProfilePath,
+             FakeProfileFinder
+           );
+           assert.equal(
+             isManuallyDefault, true,
+             'Manually configured default profile'
+           );
+
+           const isNotDefault = await firefox.isDefaultProfile(
+             path.join(profilesDirPath, 'unkown-profile-dir'),
+             FakeProfileFinder
+           );
+           assert.equal(
+             isNotDefault, false,
+             'Unknown profile path'
+           );
+         });
+       });
+
+    it('allows profile path if there is no profiles.ini file',
+       async () => {
+         return withTempDir(async (tmpDir) => {
+           const profilesDirPath = tmpDir.path();
+           const FakeProfileFinder = createFakeProfileFinder(profilesDirPath);
+
+           const isNotDefault = await firefox.isDefaultProfile(
+             '/tmp/my-custom-profile-dir',
+             FakeProfileFinder
+           );
+
+           assert.equal(isNotDefault, false);
+         });
+       });
+
+    it('rejects on any unexpected error while looking for profiles.ini',
+       async () => {
+         return withTempDir(async (tmpDir) => {
+           const profilesDirPath = tmpDir.path();
+           const FakeProfileFinder = createFakeProfileFinder(profilesDirPath);
+           const fakeFsStat = sinon.spy(() => {
+             return Promise.reject(new Error('Fake fs stat error'));
+           });
+
+           let exception;
+           try {
+             await firefox.isDefaultProfile(
+               '/tmp/my-custom-profile-dir',
+               FakeProfileFinder,
```

Comment:
I prefer to see optional function parameters passed as an object, like:
```js
firefox.isDefaultProfile('/tmp/my-custom-profile-dir', {
  profileFinder: FakeProfileFinder,
  fsStat: fakeFsStat,
}); 
```
However, this preference was born in a world without Flow so I can see how it's not such a big deal anymore. Anyway, I just wanted to point it out since it would be the first time we have a function in web-ext that does optional params like this.